esp-hal-common/src/interrupt/riscv.rs

```diff
@@ -565,6 +566,36 @@ mod vectored {
 /// # Safety
 ///
 /// This function is called from an assembly trap handler.
+/// # Safety
+///
+/// This function is called from an assembly trap handler.
+#[cfg(not(esp32c6))]
+#[doc(hidden)]
+#[export_name = "set_prio"]
+unsafe fn handle_priority()->u32{
+    let interrupt_id:usize = mcause::read().bits() & 0x0FFFFFFF; //MSB is whether its exception or interrupt.
+    let intr = &*crate::peripherals::INTERRUPT_CORE0::PTR;
+    let interrupt_priority =  intr.cpu_int_pri_0.as_ptr().offset(interrupt_id as isize).read_volatile();
+    let prev_interrupt_priority = intr.cpu_int_thresh.read().bits();
+    intr.cpu_int_thresh.write(|w| w.bits(interrupt_priority + 1)); //set the prio threshold to 1 more than current interrupt prio
```

Comment:
This won't behave well when interrupt_priority is already 15; I don't remember exactly what happened (i.e. whether the 5th+ bits were dropped and we reset the threshold to zero, or whether the chip raised an exception writing to a "reserved" bit), but that case needs special handling.

Maybe the simplest option here is to just not re-enable interrupts if we're already handling a maximum-priority interrupt?

Reply:
Agreed, i'll add an edge case for the max prio case.